### PR TITLE
Convert face crops to RGB before embedding

### PR DESCRIPTION
**Pull request by JoeApo108**
_Please read the [contributing guidelines](https://github.com/blakeblackshear/frigate/blob/dev/CONTRIBUTING.md) before submitting a PR._

### Problem
Face crops are OpenCV BGR arrays. `_process_image()` hands the ndarray straight to PIL, so both the FaceNet and ArcFace embedders receive BGR while they expect from training data the RGB order.

The error applies to enrollment and recognition alike, so it partially cancels out, which is why it went unnoticed, but it still costs accuracy.

### Fix
Convert BGR -> RGB for ndarray inputs in both `_preprocess_inputs` methods.(`custom_classification.py` already handles channel order explicitly; the face path predates it.)

### Results
Measured on a 2-person library + 29 low-quality camera crops:

| model | person separation (RGB) | (BGR) |
|---|---|---|
| arcface (large) | 0.501 | 0.480 |
| facenet (small, default) | 0.464 | 0.382 |

For facenet, cross-person similarity drops from 0.154 to 0.100 in fewer wrong-person matches.

Local tests: 781/785 pass, the 4 failures are the known go2rtc environment errors, identical on clean `dev`.

### Migration
None. Embeddings are rebuilt from the stored crops at startup.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code
- [ ] Documentation Update

## Additional information

- This PR is related to issue:

## AI disclosure

- [ ] No AI tools were used in this PR.
- [x] AI tools were used in this PR. Details below:

**AI tool(s) used**: Claude (Claude Code)

**How AI was used**: investigation, code generation, validation benchmarking, debugging

**Extent of AI involvement**: identified the channel-order mismatch, generated the fix, built/ran the validation benchmarks (RGB vs BGR A/B on a real face library with both embedding models)

**Human oversight**: I directed and reviewed all changes; validation ran on my own NVR data with personally verified ground truth.

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I can explain every line of code in this PR if asked.
- [ ] UI changes including text have used i18n keys and have been added to the `en` locale.
- [x] The code has been formatted using Ruff (`ruff format frigate`)

### CODE REVIEW

**Comment by hawkeye217** [comment]
See https://github.com/blakeblackshear/frigate/pull/23123

**Comment by JoeApo108** [comment]
@hawkeye217 

Thx for that PR link, but looking at it, It's the same kind of bug but one step further down the pipeline. #23123 fixed the detector side (YuNet wants BGR); this one fixes the embedder side (ArcFace/FaceNet want RGB). My branch is on top of that fix, so they should work together.

**Comment by hawkeye217** [comment]
RGB vs BGR seems to be a little difficult to pin down, the trap is that a quoted "channel order" is almost always the *loader's input convention*, not what the network actually consumes. The standard ArcFace loaders take a BGR image but swap it to RGB before the net.

A few pertinent links:

- insightface `get_feat`: `cv2.dnn.blobFromImages(..., swapRB=True)`: https://github.com/deepinsight/insightface/blob/master/python-package/insightface/model_zoo/arcface_onnx.py
- OpenVINO, same ResNet100 ArcFace: converted with `--reverse_input_channels`, which is the only reason its model card reads "BGR": https://github.com/openvinotoolkit/open_model_zoo/blob/master/models/public/face-recognition-resnet100-arcface-onnx/model.yml

So the *net* is RGB; the "BGR" that is quoted is the pre-swap input. FaceNet (inception_resnet_v1) is RGB too. YuNet is the genuine exception, `cv2.FaceDetectorYN` wants BGR, and the opencv_zoo demo feeds `cv2.imread`
straight in with no swap: https://github.com/opencv/opencv_zoo/blob/main/models/face_detection_yunet/demo.py 

So #23123 is correct.

I ran a quick test and verified on the models that Frigate ships (arcface.onnx / facenet.onnx) on over 400 face pairs.
RGB beats BGR on both, but not by much:

ArcFace AUC 0.999 vs 0.998 (acc 99.5 vs 99.3%),
FaceNet AUC 0.999 vs 0.997 (acc 98.3 vs 97.8%). 

This is small because the error is self-consistent across enroll+query, but RGB is indeed the correct order.

I'd suggest simplifying your PR and adding a helper to make it clearer what's being done, eg:

```
# BaseEmbedding
@staticmethod
def _bgr_to_rgb(frame):
    if isinstance(frame, np.ndarray) and frame.ndim == 3:
        return np.ascontiguousarray(frame[:, :, ::-1])
    return frame

pil = self._process_image(self._bgr_to_rgb(raw_inputs[0]))
```

**Comment by JoeApo108** [comment]
@hawkeye217 

Done, per your suggestion, I've moved it into a `BaseEmbedding._bgr_to_rgb` helper

**Review by NickM-27 — APPROVED** [review]
